src/Chat/Client/Actions/PostMessageAction.php

```diff
@@ -28,9 +28,9 @@ public function __construct(
         $this->text = $text;
     }
 
-    public function getMaxAttempts(): int
+    public function getExceptionClassName(): string
     {
-        return 5;
+        return MessagePostFailureException::class;
```

Comment:
For consistency mostly. `postMessage()` throws that exception, the alternative is the generic `ActionExecutionFailureException` which is really exposing an implementation detail unnecessarily.